apps/wallet/src/composables/external-canisters.composable.ts

```diff
@@ -0,0 +1,170 @@
+import { Principal } from '@dfinity/principal';
+import { ComputedRef, computed, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { LocationQuery, useRouter } from 'vue-router';
+import { useAutocomplete } from '~/composables/autocomplete.composable';
+import logger from '~/core/logger.core';
+import { useStationStore } from '~/stores/station.store';
+import { SelectItem } from '~/types/helper.types';
+import { ExternalCanisterStateEnum } from '~/types/station.types';
+
+export type ExternalCanistersFilters = {
+  name_prefix: string;
+  canisters: string[];
+  canister_items: SelectItem<string>[];
+  labels: string[];
+  states: ExternalCanisterStateEnum[];
+  sort_by: string;
+};
+
+export type ExternalCanistersFiltersStorable = {
```

Comment:
some of these fields are checked if they are arrays, maybe reflect this in the type definition?